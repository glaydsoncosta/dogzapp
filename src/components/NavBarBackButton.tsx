import React from 'react';
import { StyleSheet, TouchableWithoutFeedback, View } from 'react-native';
import FastImage from 'react-native-fast-image';
import { assets } from '../assets';
import { primary } from '../styles/colors';

type NavBarBackButtonProps = {
  onPress: () => void;
};

export default function NavBarBackButton(props: NavBarBackButtonProps) {
  const { onPress } = props;
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.mainContainer}>
        <FastImage source={assets.icons.misc.arrowLeft} resizeMode={FastImage.resizeMode.contain} style={styles.favoriteIcon} tintColor={primary} />
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  favoriteIcon: {
    width: 22,
    height: 22
  }
});
