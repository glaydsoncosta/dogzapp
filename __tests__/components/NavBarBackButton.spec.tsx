import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavBarBackButton from '../../src/components/NavBarBackButton';

test('Should render NavBarBackButton component', () => {
  const snapShot = renderer.create(<NavBarBackButton onPress={() => {}} />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
