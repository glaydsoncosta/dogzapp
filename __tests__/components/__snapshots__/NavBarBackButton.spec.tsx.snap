// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NavBarBackButton component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 22,
          "width": 22,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "testUri": "../../../src/assets/icons/arrow-left.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      tintColor="#673AB7"
    />
  </View>
</View>
`;
